### ISSUE TITLE
3.0.13 Build fails with libcap 2.29+

### ISSUE THREAD

**Issue by joejulian** [cyrusimap/cyrus-imapd#2974]
```
lib/util.c:582:12: error: conflicting types for 'cap_setuid'
  582 | static int cap_setuid(int uid, int is_master)
      |            ^~~~~~~~~~
In file included from lib/util.c:57:
/usr/include/sys/capability.h:138:12: note: previous declaration of 'cap_setuid' was here
  138 | extern int cap_setuid(uid_t uid);
      |            ^~~~~~~~~~
make[2]: *** [Makefile:5747: lib/lib_libcyrus_min_la-util.lo] Error 1
```

https://github.com/cyrusimap/cyrus-imapd/blob/95bfb40c041704158be48758d8ab3782caf1f980/lib/util.c#L582

**Comment by joejulian** [comment]
This is caused by this change:
https://git.kernel.org/pub/scm/libs/libcap/libcap.git/commit/?id=3165de92058397670409dac6091f9358f2772328

**Comment by elliefm** [comment]
So, if I understand this correctly, libcap has added a new public API function `cap_setuid()` that conflicts with the one cyrus already has internally?

In which case, we should be able to fix this in a simple way by just renaming our own cap_setuid (and probably anything else we implement in the cap_ namespace) to something else (cyrus_cap_foo I guess)?

It might also be worth taking a deeper look at whether we should start using these new APIs.  But for now, keep-doing-what-we're-doing-with-a-different-name is unintrusive enough that I would include it in a 3.0 patch.

Do you want to submit a pull request?  I can fix it myself easily enough, but I can't easily confirm it actually fixes the problem, so if you can do it, that'd be great.

**Comment by joejulian** [comment]
Sure... let's see how rusty my C is. :)

**Comment by elliefm** [comment]
For my own reference later, it looks like [libcap 2.29](https://sites.google.com/site/fullycapable/release-notes-for-libcap/pre-releasenotesfor229) was the first release to include the conflicting functions

**Comment by elliefm** [comment]
This is fixed on master, and will be fixed in 3.2.0-beta2

I'm not sure if I'll pick it back to 3.0, since I guess most people running a stable Cyrus release aren't running a bleeding-edge libcap.  But the patch is easy enough to apply if they do need it.

**Comment by joejulian** [comment]
Thanks! Sorry I wasn't able to find the time to contribute. :(

**Comment by elliefm** [comment]
Oh wow, I just realised the person who made the pull request was not the person who opened this issue.  So, maybe cyrus-3.0 + libcap-2.29 is common enough to just fix it on 3.0 as well (it's definitely unintrusive-enough).  I'll cherry-pick it to the branch, and it'll be included in the next 3.0 patch, whenever that is.
